Lib/datetime.py

```diff
@@ -381,6 +405,40 @@ def _check_utc_offset(name, offset):
                          "-timedelta(hours=24) and timedelta(hours=24)" %
                          (name, offset))
 
+
+def _check_int_field(value):
+    if isinstance(value, int):
+        return value
+    if isinstance(value, float):
+        raise TypeError('integer argument expected, got float')
+    try:
+        value = value.__index__()
```

Comment:
The proper way to make this call is with `operator.index` (there's a subtlety where `operator.index` will directly look up the special method on the class, while directly invoking `.__index__()` will allow instances to override it, which is not how special methods should work in general).

Reply:
If you need to *try* to use `__index__`, and only fail when it exists but returns an invalid type (non-`int`) (ignoring cases where it does not exist), do `value = type(value).__index__(value)` to match the special lookup rules.